plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/DebugNativeAppEngineStandardProjectTest.java

```diff
@@ -127,6 +127,14 @@ public void testDebugLaunch() throws Exception {
     assertEquals("Hello App Engine!",
         getUrlContents(new URL("http://localhost:8080/hello"), (int) SWTBotPreferences.TIMEOUT));
 
+    // Ensure debugger has connected by looking for well-known thread
+    assertNotNull(
+        SwtBotTreeUtilities.findItem(
```

Comment:
Can't make Kokoro Windows pass this. Saw 3 failures in a row.

```
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 21.497 s <<< FAILURE! - in com.google.cloud.tools.eclipse.integration.appengine.DebugNativeAppEngineStandardProjectTest
testDebugLaunch(com.google.cloud.tools.eclipse.integration.appengine.DebugNativeAppEngineStandardProjectTest)  Time elapsed: 20.779 s  <<< ERROR!
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at com.google.cloud.tools.eclipse.integration.appengine.DebugNativeAppEngineStandardProjectTest.testDebugLaunch(DebugNativeAppEngineStandardProjectTest.java:132)
```

Reply:
Stoopid trimStackTrace.

Reply:
```
2019-06-24T02:03:58.783738601Z  testDebugLaunch(com.google.cloud.tools.eclipse.integration.appengine.DebugNativeAppEngineStandardProjectTest)  Time elapsed: 10.976 s  <<< ERROR!
2019-06-24T02:03:58.783738601Z  java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
2019-06-24T02:03:58.783738601Z  at java.util.ArrayList.rangeCheck(ArrayList.java:657)
2019-06-24T02:03:58.783738601Z  at java.util.ArrayList.get(ArrayList.java:433)
2019-06-24T02:03:58.783738601Z  at org.eclipse.swtbot.swt.finder.finders.UIThreadRunnable.syncExec(UIThreadRunnable.java:174)
2019-06-24T02:03:58.783738601Z  at org.eclipse.swtbot.swt.finder.widgets.AbstractSWTBot.syncExec(AbstractSWTBot.java:560)
2019-06-24T02:03:58.783738601Z  at org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem.getItems(SWTBotTreeItem.java:804)
2019-06-24T02:03:58.783738601Z  at com.google.cloud.tools.eclipse.swtbot.SwtBotTreeUtilities.findItem(SwtBotTreeUtilities.java:249)
2019-06-24T02:03:58.783738601Z  at com.google.cloud.tools.eclipse.integration.appengine.DebugNativeAppEngineStandardProjectTest.testDebugLaunch(DebugNativeAppEngineStandardProjectTest.java:132)
```

Reply:
```
20:23:22.871 [Worker-2: Starting App Engine Standard at localhost] INFO  c.g.c.t.a.o.DevAppServerRunner - submitting command: C:\Program Files\Java\jdk1.8.0_152\bin\java.exe -Xdebug -Xrunjdwp:transport=dt_socket,server=n,suspend=y,quiet=y,address=49968 -Duse_jetty9_runtime=true -D--enable_all_permissions=true -Dappengine.sdk.root=C:\Program Files (x86)\Google\Cloud SDK\google-cloud-sdk\platform\google_appengine\google\appengine\tools\java -cp C:\Program Files (x86)\Google\Cloud SDK\google-cloud-sdk\platform\google_appengine\google\appengine\tools\java\lib\appengine-tools-api.jar com.google.appengine.tools.development.DevAppServerMain --address=localhost --port=8080 --allow_remote_shutdown --disable_update_check --no_java_agent T:\src\github\google-cloud-eclipse\plugins\com.google.cloud.tools.eclipse.integration.appengine\target\work\data\.metadata\.plugins\org.eclipse.wst.server.core\tmp0\testapp_java8
20:23:22.886 [Worker-2: Starting App Engine Standard at localhost] DEBUG c.g.c.t.e.a.l.s.LocalAppEngineServerBehaviour - New Process: java.lang.ProcessImpl@7e4044b4
findItem: checking top-level item: App Engine Standard at localhost [App Engine Local Server]
findItem: checking child item: Google Cloud SDK Dev App Server [251.0.0]
findItem: checking child item: Java HotSpot(TM) 64-Bit Server VM<00:00.000>[49968]
findItem: checking child item: 
findItem: checking child item: 
findItem: checking child item: 
findItem: checking child item: 
findItem: checking child item: 
findItem: checking child item: 
findItem: checking child item: Thread [qtp447981768-27] (Running)
findItem: checking child item: Thread [qtp447981768-28] (Running)
findItem: checking child item: Thread [qtp447981768-29] (Running)
findItem: checking child item: Thread [qtp447981768-30] (Running)
findItem: checking child item: Thread [qtp447981768-31] (Running)
findItem: checking child item: Thread [qtp447981768-32] (Running)
findItem: checking child item: Thread [qtp447981768-33] (Running)
findItem: checking child item: Thread [qtp447981768-34] (Running)
findItem: checking child item: Thread [qtp447981768-35] (Running)
findItem: checking child item: Thread [qtp447981768-36] (Running)
findItem: checking child item: Thread [qtp447981768-37] (Running)
findItem: checking child item: Thread [qtp447981768-38] (Running)
findItem: checking child item: Thread [qtp447981768-39] (Running)
findItem: checking child item: Thread [qtp447981768-40] (Running)
findItem: checking child item: Thread [qtp447981768-41] (Running)
findItem: checking child item: Thread [qtp447981768-42] (Running)
findItem: checking child item: Thread [qtp447981768-43] (Running)
findItem: checking child item: Thread [qtp447981768-44] (Running)
findItem: checking child item: Thread [qtp447981768-45] (Running)
findItem: checking child item: Thread [qtp447981768-46] (Running)
findItem: checking child item: Thread [qtp447981768-47] (Running)
findItem: checking child item: Thread [qtp447981768-48] (Running)
findItem: checking child item: Thread [qtp447981768-49] (Running)
findItem: checking child item: Thread [qtp447981768-50] (Running)
findItem: checking child item: Thread [qtp447981768-51] (Running)
findItem: checking child item: Thread [qtp447981768-52] (Running)
findItem: checking child item: Thread [qtp447981768-53] (Running)
findItem: checking child item: Thread [qtp447981768-54] (Running)
findItem: checking child item: Thread [qtp447981768-55] (Running)
findItem: checking child item: Thread [qtp447981768-56] (Running)
findItem: checking child item: Thread [qtp447981768-57] (Running)
findItem: checking child item: Thread [qtp447981768-58] (Running)
findItem: checking child item: Thread [qtp447981768-59] (Running)
findItem: checking child item: Daemon Thread [Scanner-0] (Running)
findItem: checking child item: Thread [qtp447981768-61] (Running)
findItem: checking child item: Thread [Connector-Scheduler-4218d6a3] (Running)
findItem: checking child item: Thread [qtp447981768-63] (Running)

!ENTRY org.eclipse.ui 4 0 2019-06-23 20:23:27.087
```

Reply:
Of course threads might die during this processing 🤦‍♂

Reply:
FYI, the recent Travis failure at 76f5ac76c2422e02353e97578307a3a2d4ec5c64 on Photon/Java 8 (https://api.travis-ci.org/v3/job/549792155/log.txt)  was at line 133

```
assertTrue(SwtBotTreeUtilities.hasChild(bot, launchTree, is("Thread [main] (Running)")))
```

so I guess it is possible that Linux may not see the `[main]` thread either. But if the test looks for any thread, we should be good.

```
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 13.603 s <<< FAILURE! - in com.google.cloud.tools.eclipse.integration.appengine.DebugNativeAppEngineStandardProjectTest
testDebugLaunch(com.google.cloud.tools.eclipse.integration.appengine.DebugNativeAppEngineStandardProjectTest)  Time elapsed: 13.595 s  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.google.cloud.tools.eclipse.integration.appengine.DebugNativeAppEngineStandardProjectTest.testDebugLaunch(DebugNativeAppEngineStandardProjectTest.java:133)
```